### PR TITLE
Calls, Recordings, Conferences, and Statistics Integration Tests

### PR DESCRIPTION
**Pull request by ckoegel**

### CODE REVIEW

**Review by ajrice6713 — APPROVED** [review]

File: spec/integration/calls_api_integration_spec.rb [hunk 1]
```diff
@@ -1,42 +1,180 @@
...
+    it 'updates a call using bxml' do
+      update_call_id = create_manteca_call(@api_instance_calls)
+      sleep(SLEEP_TIME_S)
+
+      update_bxml = "<?xml version=\"1.0\" encoding=\"UTF-8\"?><Bxml><SpeakSentence locale=\"en_US\" gender=\"female\" voice=\"susan\">This is a test bxml response</SpeakSentence><Pause duration=\"3\"/></Bxml>";
+      update_response = @api_instance_calls.update_call_bxml_with_http_info(BW_ACCOUNT_ID, update_call_id, update_bxml)
+      expect(update_response[CODE]).to eq(204)
+      sleep(SLEEP_TIME_S)
+      
+      complete_response = @api_instance_calls.update_call_with_http_info(BW_ACCOUNT_ID, update_call_id, $complete_call_body)
```
Comment (matthewkmartin) [inline spec/integration/calls_api_integration_spec.rb:115]: Even though we are updating the call to be complete, should we still add this call id to our list and verify that it is cleaned up later?

File: spec/integration/conferences_api_integration_spec.rb [hunk 2]
```diff
@@ -1,77 +1,162 @@
...
+  before(:all) do
+    Bandwidth.configure do |config|
+      config.username = BW_USERNAME
+      config.password = BW_PASSWORD
+      config.return_binary_data = true
+    end
+    @api_instance_conferences = Bandwidth::ConferencesApi.new
+    @api_instance_calls = Bandwidth::CallsApi.new
+    $manteca_test_id = setup_manteca('conference')
+    $manteca_call_id = create_manteca_call($manteca_test_id, "/bxml/joinConferencePause", @api_instance_calls)
```
Comment (matthewkmartin) [inline spec/integration/conferences_api_integration_spec.rb:16]: Should we add this call ID to the cleanup list?

File: spec/integration/recordings_api_integration_spec.rb [hunk 3]
```diff
@@ -1,84 +1,218 @@
...
+  before(:all) do
+    Bandwidth.configure do |config|
+      config.username = BW_USERNAME
+      config.password = BW_PASSWORD
+      config.return_binary_data = true
+    end
+    @api_instance_recordings = Bandwidth::RecordingsApi.new
+    @api_instance_calls = Bandwidth::CallsApi.new
+    $manteca_test_id = setup_manteca('CALL')
+    $manteca_call_id = create_manteca_call($manteca_test_id, "/bxml/startLongRecording", @api_instance_calls)
```
Comment (matthewkmartin) [inline spec/integration/recordings_api_integration_spec.rb:16]: Should we add this to a cleanup list?

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -36,19 +36,55 @@
...
+    begin
+      $active_calls.each do |call_id|
+        response = calls_api.get_call_state(BW_ACCOUNT_ID, call_id)
+        if !(response.state == 'disconnected')
+          calls_api.update_call(BW_ACCOUNT_ID, call_id, $complete_call_body)
+        end
+      end
+    rescue Bandwidth::ApiError => e
+      puts e.inspect
+      error_message = "Failed to terminate all calls" + $active_calls.to_s
```
Comment (matthewkmartin) [inline spec/spec_helper.rb:85]: Do we want to do a polling retry behavior?  This was actually a somewhat common failure for "call with id <id>" was not found in the python sdk.  What are the consequences of a call not being terminated?

**Review by matthewkmartin — CHANGES_REQUESTED** [review]

File: spec/integration/calls_api_integration_spec.rb (hunk 1 above)
Comment (ckoegel) [reply spec/integration/calls_api_integration_spec.rb:115]: All of the calls that get created using the `create_manteca_call` get added to the list of active calls within that function. [Here](https://github.com/Bandwidth/ruby-sdk/pull/71/files#diff-4bab836705b9ad756b0d0299b05d329d01b52b7bdab59781c13fac9af9eca1c7R31) is where I do it.

**Review by matthewkmartin — APPROVED** [review]

**Pull request merged**